### PR TITLE
Py3ews - Activated the legacy section in the openssl.cnf

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Status
Ready (Reason for hold)

## Related Content Pull Request
Related PR: link to the PR at demisto/content

## Related Issues
Two bugs:
Related: [XSUP-20359](https://jira-hq.paloaltonetworks.local/browse/XSUP-20359),
Related: [XSUP-20247](https://jira-hq.paloaltonetworks.local/browse/XSUP-20247).

An old bug that I think is also related: 
[XSUP-19122](https://jira-hq.paloaltonetworks.local/browse/XSUP-19122).

## Description
In the above mention bugs, EWS Mail Sender integration failed after upgrading the docker image to: demisto/py3ews:1.0.0.43044. 
The error in the logs is: 
```
  File "/usr/local/lib/python3.10/hashlib.py", line 123, in __get_builtin_constructor
    raise ValueError('unsupported hash type ' + name)
ValueError: unsupported hash type md4
```
It appears that some hashing algorithms with MD4 among them were disabled as obsoleted, which caused that error. 
To solve that we activated the legacy section in the openssl.cnf file (re-enabled md4 to hashlib) - based on the solution described here: https://github.com/ecederstrand/exchangelib/issues/608.

### CODE REVIEW

**Comment by ShacharKidor** [comment]
@JudahSchwartz - I'm not sure why the verify.py failed - by testing locally from inside the docker it passes, and also I used the created dev/demisto image in the integration itself and it seems that it solved the bug. 
Any idea?

**Review by glicht — APPROVED** [review]

**Comment by glicht** [comment]
@ShacharKidor you modified the file: docker/crypto/verify.py (in the **crypto** dir). You want to modify: **docker/py3ews/verify.py**.

File: docker/py3ews/Dockerfile [hunk 1]
```diff
@@ -5,4 +5,14 @@ COPY requirements.txt .
...
-  && apk del .build-dependencies
\ No newline at end of file
+  && apk del .build-dependencies
+
+# Handling the issue described here: https://github.com/ecederstrand/exchangelib/issues/608
+# by activating the legacy sect in openssl.cnf (re-enableing md4 to hashlib)
+RUN cp /etc/ssl/openssl.cnf /etc/ssl/openssl.cnf.org \
+&& sed -i '/default = default_sect/a legacy = legacy_sect' /etc/ssl/openssl.cnf \
+&& echo -e 'activate = 1\n\n[legacy_sect]\nactivate = 1\n' > /tmp/ssl.cnf \
+&& sed -i '/ activate = 1/r /tmp/ssl.cnf' /etc/ssl/openssl.cnf \
```
Comment (JudahSchwartz) [inline docker/py3ews/Dockerfile:15]: where is this original
`activate = 1` ?

**Review by JudahSchwartz — APPROVED** [review]
Please attach a final version of the file to the pr?

File: docker/py3ews/Dockerfile (hunk 1 above)
Comment (ShacharKidor) [reply docker/py3ews/Dockerfile:15]: I deleted it in line 16 (if that is what you are asking).

Comment (JudahSchwartz) [reply docker/py3ews/Dockerfile:15]: in line 15 you replace activate=1 with the contents of the  /tmp/ssl.cnf file, which is activate = 1\n\n[legacy_sect]\nactivate = 1\n

where is the original activate = 1 that youre appending [legacy_sect]\nactivate = 1 to?

**Pull request merged**